### ISSUE TITLE
Blocking client basic_consume() can result in dispatch of callback

### ISSUE THREAD

**Issue by happybob007** [pika/pika#626]
Using Pika version 0.9.14 on Linux (Ubuntu Mint 17, 64-bit).

In at least some scenarios, calling `basic_consume()` on a `BlockingChannel` can result in the callback you are registering being called immediately.

That may be a bit unexpected -- it is for me.

Here's a stack trace that might be useful.  Sadly, I haven't spent more than a couple of minutes looking at it.

```
  File "build/bdist.linux-x86_64/egg/diagonal/mqueue/pikawrapper.py", line 312, in _try_to_subscribe
    self.channel.basic_consume(callback, queue=queue_name, no_ack=no_ack)
  File "build/bdist.linux-x86_64/egg/pika/channel.py", line 211, in basic_consume
    {'consumer_tag': consumer_tag})])
  File "build/bdist.linux-x86_64/egg/pika/adapters/blocking_connection.py", line 954, in _rpc
    self._wait_on_response(method_frame))
  File "build/bdist.linux-x86_64/egg/pika/adapters/blocking_connection.py", line 980, in _send_method
    self.connection.process_data_events()
  File "build/bdist.linux-x86_64/egg/pika/adapters/blocking_connection.py", line 129, in process_data_events
    if self._handle_read():
  File "build/bdist.linux-x86_64/egg/pika/adapters/blocking_connection.py", line 232, in _handle_read
    super(BlockingConnection, self)._handle_read()
  File "build/bdist.linux-x86_64/egg/pika/adapters/base_connection.py", line 307, in _handle_read
    self._on_data_available(data)
  File "build/bdist.linux-x86_64/egg/pika/connection.py", line 1145, in _on_data_available
    self._process_frame(frame_value)
  File "build/bdist.linux-x86_64/egg/pika/connection.py", line 1200, in _process_frame
    self._deliver_frame_to_channel(frame_value)
  File "build/bdist.linux-x86_64/egg/pika/connection.py", line 848, in _deliver_frame_to_channel
    return self._channels[value.channel_number]._handle_content_frame(value)
  File "build/bdist.linux-x86_64/egg/pika/channel.py", line 775, in _handle_content_frame
    self._on_deliver(*response)
  File "build/bdist.linux-x86_64/egg/pika/channel.py", line 852, in _on_deliver
    body)
  File "build/bdist.linux-x86_64/egg/diagonal/diagonalclient.py", line 368, in pika_cb
    return callback(de, self, user_data)
  File "./signuprequest.py", line 143, in new_signupRequest_cb
```

**Comment by vitaly-krugl** [comment]
@happybob007: this is a know issue with the legacy blocking client. The blocking client has been completely rewritten, and this should no longer be an issue in version 0.10.0. The prerelease version of 0.10.0 is on PyPi already - please give it a try. You should see a noticeable performance boost there as well, particularly with `basic_publish`. I install the prerelease on my MabBook like this: `pip install --pre --user pika>=0.10.0`.

**Comment by vitaly-krugl** [comment]
Fixed in pika 0.10.0. Please close this issue.

**Comment by happybob007** [comment]
Closing
